### ISSUE TITLE
Bug: Auto-Generated YouTube Videos Don't Work

### ISSUE THREAD

**Issue by Aurorum** [Just-Some-Bots/MusicBot#1818]
Which version are you using?

The bot is not mine. 

What type of issue are you creating?

- [ ✅ ] Bug
- [ ] Feature request
- [ ] Question

----

### Description of issue

Playing an auto-generated song from YouTube provokes an error and doesn't play anything.

### Steps to reproduce

1. Find an auto-generated song on YouTube (it'll say it's auto-generated in the description, [When Doves Cry](https://youtu.be/oTUSeac7IuQ) is an example)
2. Do a command '!play (song URL)'
3. See error

![screenshot_20190113-123333](https://user-images.githubusercontent.com/43215253/51089795-28811b00-176b-11e9-812f-372afdbf16d3.jpg)

### Log file

This is not my bot, so it would not be possible to provide this (apologies).

**Comment by AutumnClove** [comment]
Okay, this is a bit tricky. If the user actually wants to play the whole playlist, he needs to pass the actual playlist URL (youtube.com/playlist?list=...) to mb, however the Youtube website doesn't provide these URLs for mixes. You can however still build such a URL yourself which will work with mb.

**Comment by Aurorum** [comment]
I'm not sure I follow: the user doesn't need to select a whole playlist to play an auto-generated video. You can select a single song, and generate a link for it (see the Prince song example above), which can be shared, but won't work with the bot. This isn't part of a generated mix or playlist.

**Comment by AutumnClove** [comment]
Single songs work with the bot because of the way links work, but to play a playlist you need the `playlist?list=` but with auto generated ones YouTube doesn't give that, which makes the list unplayable, you could queue one song at a time as a different method, or create your own playlist with the same music, and use that link, however other than that, I don't know if support will be added for mixes as they don't have a normal playlist link.
Tl:Dr YouTube mixes don't have the right link to be played as a playlist.
Also the single songs you linked worked just fine for me.

**Comment by AutumnClove** [comment]
Oh I've misread this whole thing, I thought by auto generated you meant YouTube mixes

**Comment by Aurorum** [comment]
Ah, yep, I'm referring to the auto-generated songs. They have it at the bottom of the description.

![screenshot_20190113-201935](https://user-images.githubusercontent.com/43215253/51090229-a1cf3c80-1770-11e9-9810-066bdd30e9f8.jpg)

It definitely seems that it doesn't work for anyone using any auto-generated song, so it's odd that there's no issues on your side. Maybe it's just the server I'm in, but a) I doubt there's anything special about it which would cause this and b) I'm not sure how to test another. :thinking:

**Comment by AutumnClove** [comment]
Are you maybe getting Geo blocked?

**Comment by Aurorum** [comment]
Affects all users in the servers and it loads fine for me when playing on YouTube. That being said, I am aware that those videos are blocked in some countries. I'm not familiar with setting up the bot, can the bot be set to a specific country, which might be causing this issue?

**Comment by AutumnClove** [comment]
It can't but it could be hosted somewhere causing it to get blocked, I'd ask about that, and try a restart, there's a chance it's just a weird glitch currently and needs a fresh start

**Comment by Aurorum** [comment]
The bot has been reset several times in the past month using the `!restart` command, if that's what you're referring to...unfortunately, that hasn't provided a fix.

**Comment by AutumnClove** [comment]
Restart doesn't fully restart your bot properly unfortunately due to the nature of it currently, for a full restart, you'll need to shut the bot down and re-run the run file.

**Comment by deceptivelyjosh** [comment]
I'm the owner of the bot pictured and this issue was linked to me by the OP, this is an issue with geo restrictions as you suggested caused by YouTube improperly detecting the server the bot is hosted on as being in UAE instead of France but that's an issue with YouTube and not MusicBot or YouTube-DL which is why I have never personally made an issue here, sorry for any confusion.

**Comment by MattBSG** [comment]
No problem, if it turns out to appear it wasn’t actually a geo-locked issue, please feel free to re-open.
